test/conftest.py

```diff
@@ -734,3 +777,157 @@ def sampleset_ref(clients: dict[str, Any], ws_id: int, reads_refs: list[str]) ->
         ws_id,
         clients["ws"],
     )
+
+
+@pytest.fixture(scope="session")
+def assembly_set(
```

Comment:
Abstracted out the save_set function so the fixtures for the sets now look a lot simpler.

Note that they use the `ws_id` param, which means you can save the set in either the `list_all_sets` or the `default` workspace. It also ensures that any fixtures that are used to create this fixture are also created using the appropriate value for the `ws_id` param. (It's turtles all the way down.)